### PR TITLE
Add pre-commit, black, isort and flake8. Big refactor of code.

### PR DESCRIPTION
**Pull request by jumcorredorro**
## Description
Add pre-commit, black, isort and flake8. 
Big refactor of code, because library suggestions.

## Changes include

- [ ] Bugfix (non-breaking change that solves an issue, bump the third digit of the version)
- [X] New feature (non-breaking change that adds functionality, bump the second digit of the version)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality, bump the first digit of the version)
- [ ] Documentation update (Don't bump the version of the project)

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    rev: 4.0.1
+    hooks:
+      - id: flake8
+        args: ['--config=setup.cfg']
+        additional_dependencies: [flake8-isort]
+
+  # - repo: https://github.com/pre-commit/mirrors-mypy
+  #   rev: v0.942
+  #   hooks:
+  #     - id: mypy
```
Comment (brunomichetti) [inline .pre-commit-config.yaml:33]: Why is this commented?

Comment (jumcorredorro) [reply .pre-commit-config.yaml:33]: I will use it in next PR, that is why, do you want to remove it?

File: drip/admin.py [hunk 2]
```diff
@@ -48,48 +45,40 @@ def timeline(self, request, drip_id, into_past, into_future):
...
             )
-            seen_users.update(
-                shifted_drip.get_queryset().values_list('id', flat=True)
-            )
+            seen_users.update(shifted_drip.get_queryset().values_list("id", flat=True))
 
-        return render(request, 'drip/timeline.html', locals())
+        return render(request, "drip/timeline.html", locals())
 
     def get_mime_html_from_alternatives(self, alternatives):
```
[the remark this answers is not in the document]
Comment (jumcorredorro) [reply drip/admin.py:62]: Yes, mypy step in pre-commit and typing will be included in other PR's in the future.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (brunomichetti) [reply .pre-commit-config.yaml:33]: Yes I think it's better to remove this from here and add it in the next PR

**Review by malexandroff1 — APPROVED** [review]
LG2M

**Pull request merged**